test/e2e/run_test.go

```diff
@@ -184,4 +184,15 @@ var _ = Describe("Podman run", func() {
 		Expect(session.ExitCode()).To(Equal(0))
 		Expect(session.OutputToString()).To(ContainSubstring("15"))
 	})
+
+	It("podman run notify_socket", func() {
+		sock := "/run/sock"
+		os.Setenv("NOTIFY_SOCKET", sock)
+		session := podmanTest.Podman([]string{"run", "--rm", ALPINE, "printenv", "NOTIFY_SOCKET"})
+		session.Wait(10)
+		Expect(session.ExitCode()).To(Equal(0))
+		match, _ := session.GrepString(sock)
+		Expect(match).Should(BeTrue())
+		os.Unsetenv("NOTIFY_SOCKET")
+	})
```

Comment:
Anyway to work up a test for LISTEN_FDS?

Reply:
You would have to open a socket, and then set the flags, I guess.  Maybe this is something we could do in the test suite.

Reply:
@TomSweeneyRedHat Actually looking at this deeper, it is very difficult to create a test for this, but doing so I found other issues.  Conmon needs to setup LISTEN_PID correctly for this to work.
https://github.com/kubernetes-incubator/cri-o/pull/1300

I think we should merge this and then attempt to setup a manual test in systemd to make sure it all works.

The way this is supposed to work is systemd will create a process, then will set the LISTEN_FDS=#PASSED in FDS and a LISTEN_PID which is equal to the PID that podman WILL execute with.  At which point systemd execs podman.  PODMAN then needs to pass these variables down to conmon. 
conmon needs to modify LISTEN_PID to the PID that it is going to launch the OCI RUntime with (runc). Then RUNC will modify LISTEN_PID one more time to PID=1, which will launch the container process.

Getting podman to run with the correct LISTEN_PID,  is something we could work with in the future, not sure how easy that would be to do with GOLANG. Much easier with bash or C.

Reply:
When I replied earlier I was trying to think of a way to test it, but couldn't concoct one off the top of my head.  Chicken and egg thing.  Thanks for taking a look into it and I'm glad you found some things that should be chased due to that investigation.